### This Notebook will be used for Capstone Project

In [2]:
import requests
import pandas as pd

url = 'https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'
html = requests.get(url).content
df_list = pd.read_html(html)
df_list
df = df_list[0]
#print(df)
#df.to_csv('my data.csv')

In [3]:
df

Postal code           Borough  \
0           M1A      Not assigned   
1           M2A      Not assigned   
2           M3A        North York   
3           M4A        North York   
4           M5A  Downtown Toronto   
5           M6A        North York   
6           M7A  Downtown Toronto   
7           M8A      Not assigned   
8           M9A         Etobicoke   
9           M1B       Scarborough   
10          M2B      Not assigned   
11          M3B        North York   
12          M4B         East York   
13          M5B  Downtown Toronto   
14          M6B        North York   
15          M7B      Not assigned   
16          M8B      Not assigned   
17          M9B         Etobicoke   
18          M1C       Scarborough   
19          M2C      Not assigned   
20          M3C        North York   
21          M4C         East York   
22          M5C  Downtown Toronto   
23          M6C              York   
24          M7C      Not assigned   
25          M8C      Not assigned   
26          M9C         Etobicoke   
27          M1E       Scarborough   
28          M2E      Not assigned   
29          M3E      Not assigned   
..          ...               ...   
150         M7W      Not assigned   
151         M8W         Etobicoke   
152         M9W         Etobicoke   
153         M1X       Scarborough   
154         M2X      Not assigned   
155         M3X      Not assigned   
156         M4X  Downtown Toronto   
157         M5X  Downtown Toronto   
158         M6X      Not assigned   
159         M7X      Not assigned   
160         M8X         Etobicoke   
161         M9X      Not assigned   
162         M1Y      Not assigned   
163         M2Y      Not assigned   
164         M3Y      Not assigned   
165         M4Y  Downtown Toronto   
166         M5Y      Not assigned   
167         M6Y      Not assigned   
168         M7Y      East Toronto   
169         M8Y         Etobicoke   
170         M9Y      Not assigned   
171         M1Z      Not assigned   
172         M2Z      Not assigned   
173         M3Z      Not assigned   
174         M4Z      Not assigned   
175         M5Z      Not assigned   
176         M6Z      Not assigned   
177         M7Z      Not assigned   
178         M8Z         Etobicoke   
179         M9Z      Not assigned   

                                          Neighborhood  
0                                                  NaN  
1                                                  NaN  
2                                            Parkwoods  
3                                     Victoria Village  
4                           Regent Park / Harbourfront  
5                    Lawrence Manor / Lawrence Heights  
6         Queen's Park / Ontario Provincial Government  
7                                                  NaN  
8                                     Islington Avenue  
9                                      Malvern / Rouge  
10                                                 NaN  
11                                           Don Mills  
12                    Parkview Hill / Woodbine Gardens  
13                            Garden District, Ryerson  
14                                           Glencairn  
15                                                 NaN  
16                                                 NaN  
17   West Deane Park / Princess Gardens / Martin Gr...  
18            Rouge Hill / Port Union / Highland Creek  
19                                                 NaN  
20                                           Don Mills  
21                                    Woodbine Heights  
22                                      St. James Town  
23                                  Humewood-Cedarvale  
24                                                 NaN  
25                                                 NaN  
26   Eringate / Bloordale Gardens / Old Burnhamthor...  
27                 Guildwood / Morningside / West Hill  
28                                                 NaN  

In [4]:
df = df[df.Borough != 'Not assigned']
df

Postal code           Borough  \
2           M3A        North York   
3           M4A        North York   
4           M5A  Downtown Toronto   
5           M6A        North York   
6           M7A  Downtown Toronto   
8           M9A         Etobicoke   
9           M1B       Scarborough   
11          M3B        North York   
12          M4B         East York   
13          M5B  Downtown Toronto   
14          M6B        North York   
17          M9B         Etobicoke   
18          M1C       Scarborough   
20          M3C        North York   
21          M4C         East York   
22          M5C  Downtown Toronto   
23          M6C              York   
26          M9C         Etobicoke   
27          M1E       Scarborough   
30          M4E      East Toronto   
31          M5E  Downtown Toronto   
32          M6E              York   
36          M1G       Scarborough   
39          M4G         East York   
40          M5G  Downtown Toronto   
41          M6G  Downtown Toronto   
45          M1H       Scarborough   
46          M2H        North York   
47          M3H        North York   
48          M4H         East York   
..          ...               ...   
111         M4R   Central Toronto   
112         M5R   Central Toronto   
113         M6R      West Toronto   
114         M7R       Mississauga   
116         M9R         Etobicoke   
117         M1S       Scarborough   
120         M4S   Central Toronto   
121         M5S  Downtown Toronto   
122         M6S      West Toronto   
126         M1T       Scarborough   
129         M4T   Central Toronto   
130         M5T  Downtown Toronto   
135         M1V       Scarborough   
138         M4V   Central Toronto   
139         M5V  Downtown Toronto   
142         M8V         Etobicoke   
143         M9V         Etobicoke   
144         M1W       Scarborough   
147         M4W  Downtown Toronto   
148         M5W  Downtown Toronto   
151         M8W         Etobicoke   
152         M9W         Etobicoke   
153         M1X       Scarborough   
156         M4X  Downtown Toronto   
157         M5X  Downtown Toronto   
160         M8X         Etobicoke   
165         M4Y  Downtown Toronto   
168         M7Y      East Toronto   
169         M8Y         Etobicoke   
178         M8Z         Etobicoke   

                                          Neighborhood  
2                                            Parkwoods  
3                                     Victoria Village  
4                           Regent Park / Harbourfront  
5                    Lawrence Manor / Lawrence Heights  
6         Queen's Park / Ontario Provincial Government  
8                                     Islington Avenue  
9                                      Malvern / Rouge  
11                                           Don Mills  
12                    Parkview Hill / Woodbine Gardens  
13                            Garden District, Ryerson  
14                                           Glencairn  
17   West Deane Park / Princess Gardens / Martin Gr...  
18            Rouge Hill / Port Union / Highland Creek  
20                                           Don Mills  
21                                    Woodbine Heights  
22                                      St. James Town  
23                                  Humewood-Cedarvale  
26   Eringate / Bloordale Gardens / Old Burnhamthor...  
27                 Guildwood / Morningside / West Hill  
30                                         The Beaches  
31                                         Berczy Park  
32                                 Caledonia-Fairbanks  
36                                              Woburn  
39                                             Leaside  
40                                  Central Bay Street  
41                                            Christie  
45                                           Cedarbrae  
46                                   Hillcrest Village  
47   Bathurst Manor / Wilson Heights / Downsview North  

In [5]:
count = df['Postal code'].value_counts()
count.loc[count>1]

Series([], Name: Postal code, dtype: int64)

In [6]:
df[df['Neighborhood']== 'Not assigned']

Empty DataFrame
Columns: [Postal code, Borough, Neighborhood]
Index: []

In [7]:
df.shape

(103, 3)

## to get longitude latitude of area

In [8]:
url2= 'http://cocl.us/Geospatial_data'

In [9]:
df2 = pd.read_csv(url2)

In [10]:
df2.columns = ['Postal code', 'Latitude', 'Longitude']

In [11]:
df.set_index(['Postal code'],inplace = True)
df2.set_index(['Postal code'], inplace = True)

In [12]:
df= df.join(df2).reset_index()
df

Postal code           Borough  \
0           M3A        North York   
1           M4A        North York   
2           M5A  Downtown Toronto   
3           M6A        North York   
4           M7A  Downtown Toronto   
5           M9A         Etobicoke   
6           M1B       Scarborough   
7           M3B        North York   
8           M4B         East York   
9           M5B  Downtown Toronto   
10          M6B        North York   
11          M9B         Etobicoke   
12          M1C       Scarborough   
13          M3C        North York   
14          M4C         East York   
15          M5C  Downtown Toronto   
16          M6C              York   
17          M9C         Etobicoke   
18          M1E       Scarborough   
19          M4E      East Toronto   
20          M5E  Downtown Toronto   
21          M6E              York   
22          M1G       Scarborough   
23          M4G         East York   
24          M5G  Downtown Toronto   
25          M6G  Downtown Toronto   
26          M1H       Scarborough   
27          M2H        North York   
28          M3H        North York   
29          M4H         East York   
..          ...               ...   
73          M4R   Central Toronto   
74          M5R   Central Toronto   
75          M6R      West Toronto   
76          M7R       Mississauga   
77          M9R         Etobicoke   
78          M1S       Scarborough   
79          M4S   Central Toronto   
80          M5S  Downtown Toronto   
81          M6S      West Toronto   
82          M1T       Scarborough   
83          M4T   Central Toronto   
84          M5T  Downtown Toronto   
85          M1V       Scarborough   
86          M4V   Central Toronto   
87          M5V  Downtown Toronto   
88          M8V         Etobicoke   
89          M9V         Etobicoke   
90          M1W       Scarborough   
91          M4W  Downtown Toronto   
92          M5W  Downtown Toronto   
93          M8W         Etobicoke   
94          M9W         Etobicoke   
95          M1X       Scarborough   
96          M4X  Downtown Toronto   
97          M5X  Downtown Toronto   
98          M8X         Etobicoke   
99          M4Y  Downtown Toronto   
100         M7Y      East Toronto   
101         M8Y         Etobicoke   
102         M8Z         Etobicoke   

                                          Neighborhood   Latitude  Longitude  
0                                            Parkwoods  43.753259 -79.329656  
1                                     Victoria Village  43.725882 -79.315572  
2                           Regent Park / Harbourfront  43.654260 -79.360636  
3                    Lawrence Manor / Lawrence Heights  43.718518 -79.464763  
4         Queen's Park / Ontario Provincial Government  43.662301 -79.389494  
5                                     Islington Avenue  43.667856 -79.532242  
6                                      Malvern / Rouge  43.806686 -79.194353  
7                                            Don Mills  43.745906 -79.352188  
8                     Parkview Hill / Woodbine Gardens  43.706397 -79.309937  
9                             Garden District, Ryerson  43.657162 -79.378937  
10                                           Glencairn  43.709577 -79.445073  
11   West Deane Park / Princess Gardens / Martin Gr...  43.650943 -79.554724  
12            Rouge Hill / Port Union / Highland Creek  43.784535 -79.160497  
13                                           Don Mills  43.725900 -79.340923  
14                                    Woodbine Heights  43.695344 -79.318389  
15                                      St. James Town  43.651494 -79.375418  
16                                  Humewood-Cedarvale  43.693781 -79.428191  
17   Eringate / Bloordale Gardens / Old Burnhamthor...  43.643515 -79.577201  
18                 Guildwood / Morningside / West Hill  43.763573 -79.188711  
19                                         The Beaches  43.676357 -79.293031  
20                                         Berczy 

# part 3 capstone project

I am taking only torronto data 

In [13]:
toronto= df[df['Borough'].str.contains('Toronto', na = False)]

In [14]:
toronto.reset_index()

index Postal code           Borough  \
0       2         M5A  Downtown Toronto   
1       4         M7A  Downtown Toronto   
2       9         M5B  Downtown Toronto   
3      15         M5C  Downtown Toronto   
4      19         M4E      East Toronto   
5      20         M5E  Downtown Toronto   
6      24         M5G  Downtown Toronto   
7      25         M6G  Downtown Toronto   
8      30         M5H  Downtown Toronto   
9      31         M6H      West Toronto   
10     36         M5J  Downtown Toronto   
11     37         M6J      West Toronto   
12     41         M4K      East Toronto   
13     42         M5K  Downtown Toronto   
14     43         M6K      West Toronto   
15     47         M4L      East Toronto   
16     48         M5L  Downtown Toronto   
17     54         M4M      East Toronto   
18     61         M4N   Central Toronto   
19     62         M5N   Central Toronto   
20     67         M4P   Central Toronto   
21     68         M5P   Central Toronto   
22     69         M6P      West Toronto   
23     73         M4R   Central Toronto   
24     74         M5R   Central Toronto   
25     75         M6R      West Toronto   
26     79         M4S   Central Toronto   
27     80         M5S  Downtown Toronto   
28     81         M6S      West Toronto   
29     83         M4T   Central Toronto   
30     84         M5T  Downtown Toronto   
31     86         M4V   Central Toronto   
32     87         M5V  Downtown Toronto   
33     91         M4W  Downtown Toronto   
34     92         M5W  Downtown Toronto   
35     96         M4X  Downtown Toronto   
36     97         M5X  Downtown Toronto   
37     99         M4Y  Downtown Toronto   
38    100         M7Y      East Toronto   

                                         Neighborhood   Latitude  Longitude  
0                          Regent Park / Harbourfront  43.654260 -79.360636  
1        Queen's Park / Ontario Provincial Government  43.662301 -79.389494  
2                            Garden District, Ryerson  43.657162 -79.378937  
3                                      St. James Town  43.651494 -79.375418  
4                                         The Beaches  43.676357 -79.293031  
5                                         Berczy Park  43.644771 -79.373306  
6                                  Central Bay Street  43.657952 -79.387383  
7                                            Christie  43.669542 -79.422564  
8                          Richmond / Adelaide / King  43.650571 -79.384568  
9                       Dufferin / Dovercourt Village  43.669005 -79.442259  
10  Harbourfront East / Union Station / Toronto Is...  43.640816 -79.381752  
11                          Little Portugal / Trinity  43.647927 -79.419750  
12                      The Danforth West / Riverdale  43.679557 -79.352188  
13          Toronto Dominion Centre / Design Exchange  43.647177 -79.381576  
14     Brockton / Parkdale Village / Exhibition Place  43.636847 -79.428191  
15                    India Bazaar / The Beaches West  43.668999 -79.315572  
16                    Commerce Court / Victoria Hotel  43.648198 -79.379817  
17                                    Studio District  43.659526 -79.340923  
18                                      Lawrence Park  43.728020 -79.388790  
19                                           Roselawn  43.711695 -79.416936  
20                                   Davisville North  43.712751 -79.390197  
21                           Forest Hill North & West  43.696948 -79.411307  
22                     High Park / The Junction South  43.661608 -79.464763  
23                                 North Toronto West  43.715383 -79.405678  
24              The Annex / North Midtown / Yorkville  43.672710 -79.405678  
25                            Parkdale / Roncesvalles  43.648960 -79.456325  
26                                         Davisville  43.704324 -79.388790  
27                    University of Toronto / Harbord  43.662696 -79.400049  
28                   

first lets create a map of toronto area 

In [15]:
tor_lat =43.6532
tor_long = -79.3832

In [17]:
!pip install folium
import folium

     |████████████████████████████████| 92kB 15.3MB/s eta 0:00:01


In [18]:
mymap = folium.Map(location = [tor_lat,tor_long], zoom_start = 10)

for lat, long, label in zip(df['Latitude'],df['Longitude'],df['Neighborhood']):
# add markers 
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat,long],
        radius = 5,
        popup = label,
        color = 'blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7).add_to (mymap)
mymap

In [19]:
CLIENT_ID = 'WWJDLUUCT0G3GCJHAXDMDNSQZRU24TAGCF3P1QCB0UYYRTG4' # your Foursquare ID
CLIENT_SECRET = 'KJUIPFAPR143FYMSKABR1FBID2ZN1WYKTHHPH4ASEM5LRRIF' # your Foursquare Secret
VERSION = '20180604'
LIMIT = 100

In [20]:
import requests 
import pandas as pd
from pandas.io.json import json_normalize

### loop all neighborhoods to find nearby venues

In [21]:
def getvenues (names, latitude, longitude, rad = 500):
    venuelist = []
    for name, lat, lng in zip(names, latitude, longitude):
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius ={}&limit ={}'.format(CLIENT_ID,CLIENT_SECRET,VERSION,lat,lng,rad,LIMIT)
        results = requests.get(url).json()
        data = json_normalize(results['response']['groups'][0]['items'])
        for i in range(0,len(data)):
            venuelist.append([name,lat,lng,data['venue.name'][i],data['venue.categories'][i][0]['name'],data['venue.location.lat'][i],data['venue.location.lng'][i]])
    print (venuelist)
    return (venuelist)

In [22]:
ven= getvenues(toronto.Neighborhood, toronto.Latitude, toronto.Longitude)
venues = pd.DataFrame(ven)

[['Regent Park / Harbourfront', 43.6542599, -79.3606359, 'Roselle Desserts', 'Bakery', 43.653446723052674, -79.3620167174383], ['Regent Park / Harbourfront', 43.6542599, -79.3606359, 'Tandem Coffee', 'Coffee Shop', 43.65355870959944, -79.36180945913513], ['Regent Park / Harbourfront', 43.6542599, -79.3606359, 'Impact Kitchen', 'Restaurant', 43.65636850543279, -79.35697968750694], ['Regent Park / Harbourfront', 43.6542599, -79.3606359, 'The Distillery Historic District', 'Historic Site', 43.65024435658077, -79.35932278633118], ['Regent Park / Harbourfront', 43.6542599, -79.3606359, 'Distillery Sunday Market', 'Farmers Market', 43.650074989330655, -79.36183171318665], ['Regent Park / Harbourfront', 43.6542599, -79.3606359, 'Corktown Common', 'Park', 43.655617799749734, -79.3562113397429], ['Regent Park / Harbourfront', 43.6542599, -79.3606359, 'Souk Tabule', 'Mediterranean Restaurant', 43.65375556880743, -79.35439006096168], ['Regent Park / Harbourfront', 43.6542599, -79.3606359, 'Cooper

In [23]:
venues.columns = ['Neighborhood','Neighborhood lat','Neighborhood long','venue name', 'venue type', 'venue lat', 'venue long']
venues.tail()

Neighborhood  Neighborhood lat  \
1165  Business reply mail Processing CentrE         43.662744   
1166  Business reply mail Processing CentrE         43.662744   
1167  Business reply mail Processing CentrE         43.662744   
1168  Business reply mail Processing CentrE         43.662744   
1169  Business reply mail Processing CentrE         43.662744   

      Neighborhood long               venue name      venue type  venue lat  \
1165         -79.321558           Greenwood Park            Park  43.669025   
1166         -79.321558       Left Field Brewery         Brewery  43.673705   
1167         -79.321558     Ashbridge's Bay Park            Park  43.661822   
1168         -79.321558  Doll Factory by Damzels        Boutique  43.662158   
1169         -79.321558        Brett's Ice Cream  Ice Cream Shop  43.667222   

      venue long  
1165  -79.328163  
1166  -79.330310  
1167  -79.310424  
1168  -79.335165  
1169  -79.312831

In [25]:
venues.shape


(1170, 7)

In [31]:
len(venues['venue type'].unique())

194

there are total 194 unique venues 

## create a data frame that has top three venues in each neighborhood and there frequency 

In [36]:
## lets find out number of venues in each neighborhood
venues.groupby(['Neighborhood']).count()

Neighborhood lat  \
Neighborhood                                                           
Berczy Park                                                       30   
Brockton / Parkdale Village / Exhibition Place                    30   
Business reply mail Processing CentrE                             30   
CN Tower / King and Spadina / Railway Lands / H...                30   
Central Bay Street                                                30   
Christie                                                          30   
Church and Wellesley                                              30   
Commerce Court / Victoria Hotel                                   30   
Davisville                                                        30   
Davisville North                                                  30   
Dufferin / Dovercourt Village                                     30   
First Canadian Place / Underground city                           30   
Forest Hill North & West                                          30   
Garden District, Ryerson                                          30   
Harbourfront East / Union Station / Toronto Isl...                30   
High Park / The Junction South                                    30   
India Bazaar / The Beaches West                                   30   
Kensington Market / Chinatown / Grange Park                       30   
Lawrence Park                                                     30   
Little Portugal / Trinity                                         30   
Moore Park / Summerhill East                                      30   
North Toronto West                                                30   
Parkdale / Roncesvalles                                           30   
Queen's Park / Ontario Provincial Government                      30   
Regent Park / Harbourfront                                        30   
Richmond / Adelaide / King                                        30   
Rosedale                                                          30   
Roselawn                                                          30   
Runnymede / Swansea                                               30   
St. James Town                                                    30   
St. James Town / Cabbagetown                                      30   
Stn A PO Boxes                                                    30   
Studio District                                                   30   
Summerhill West / Rathnelly / South Hill / Fore...                30   
The Annex / North Midtown / Yorkville                             30   
The Beaches                                                       30   
The Danforth West / Riverdale                                     30   
Toronto Dominion Centre / Design Exchange                         30   
University of Toronto / Harbord                                   30   

                                                    Neighborhood long  \
Neighborhood                                                            
Berczy Park                                                        30   
Brockton / Parkdale Village / Exhibition Place                     30   
Business reply mail Processing CentrE                              30   
CN Tower / King and Spadina / Railway Lands / H...                 30   
Central Bay Street                                                 30   
Christie                                                           30   
Church and Wellesley                                               30   
Commerce Court / Victoria Hotel                                    30   
Davisville                                                         30   
Davisville North                                                   30   
Dufferin / Dovercourt Village                                      30   
First Canadian Place / Underground city                            30   
Forest Hill North & West                                           30   
Gard

In [83]:
#neigh = list(toronto.Neighborhood)[38]
temp = venues[venues['Neighborhood']==neigh]['venue type']
freqtop = list(temp.value_counts())[0:3]
top = list(temp.value_counts().index)[0:3]
dfven=[]
for neigh in list(toronto.Neighborhood):
    temp = venues[venues['Neighborhood']==neigh]['venue type']
    freqtop = list(temp.value_counts())[0:3]
    top = list(temp.value_counts().index)[0:3]
    dfven.append([neigh]+top+freqtop)
dfven
    

[['Regent Park / Harbourfront',
  'Coffee Shop',
  'Park',
  'Breakfast Spot',
  6,
  3,
  2],
 ["Queen's Park / Ontario Provincial Government",
  'Coffee Shop',
  'Sushi Restaurant',
  'Italian Restaurant',
  7,
  3,
  2],
 ['Garden District, Ryerson', 'Coffee Shop', 'Theater', 'Café', 3, 2, 2],
 ['St. James Town', 'Gastropub', 'Café', 'Farmers Market', 3, 3, 2],
 ['The Beaches', 'Beach', 'Breakfast Spot', 'BBQ Joint', 4, 3, 2],
 ['Berczy Park', 'Cocktail Bar', 'Coffee Shop', 'Seafood Restaurant', 2, 2, 2],
 ['Central Bay Street', 'Coffee Shop', 'Café', 'Bar', 8, 2, 1],
 ['Christie', 'Café', 'Korean Restaurant', 'Cocktail Bar', 3, 2, 2],
 ['Richmond / Adelaide / King',
  'Coffee Shop',
  'Café',
  'American Restaurant',
  3,
  3,
  2],
 ['Dufferin / Dovercourt Village',
  'Café',
  'Italian Restaurant',
  'Coffee Shop',
  8,
  4,
  3],
 ['Harbourfront East / Union Station / Toronto Islands',
  'Park',
  'Plaza',
  'Hotel',
  2,
  2,
  2],
 ['Little Portugal / Trinity',
  'Bar',
  'Asi

In [84]:
df2  = pd.DataFrame(dfven)
df2.columns = ['Neighborhood','1st most common venue', '2nd most common venue','3rd most common venue','frequency of first venue ','frequesncy of 2nd venue','frequency of third venue']

In [85]:
df2.head()

Neighborhood 1st most common venue  \
0                    Regent Park / Harbourfront           Coffee Shop   
1  Queen's Park / Ontario Provincial Government           Coffee Shop   
2                      Garden District, Ryerson           Coffee Shop   
3                                St. James Town             Gastropub   
4                                   The Beaches                 Beach   

  2nd most common venue 3rd most common venue  frequency of first venue   \
0                  Park        Breakfast Spot                          6   
1      Sushi Restaurant    Italian Restaurant                          7   
2               Theater                  Café                          3   
3                  Café        Farmers Market                          3   
4        Breakfast Spot             BBQ Joint                          4   

   frequesncy of 2nd venue  frequency of third venue  
0                        3                         2  
1                        3                         2  
2                        2                         2  
3                        3                         2  
4                        3                         2

## use kmeans for clustering above data frame 

In [86]:
from sklearn.cluster import KMeans
km = KMeans(init = "k-means++", n_clusters = 5, n_init = 12)

In [89]:
dfcluster= df2.iloc[:,4:7]
dfcluster.head()

frequency of first venue   frequesncy of 2nd venue  \
0                          6                        3   
1                          7                        3   
2                          3                        2   
3                          3                        3   
4                          4                        3   

   frequency of third venue  
0                         2  
1                         2  
2                         2  
3                         2  
4                         2

In [90]:
km.fit(dfcluster)

KMeans(algorithm='auto', copy_x=True, init='k-means++', max_iter=300,
    n_clusters=5, n_init=12, n_jobs=None, precompute_distances='auto',
    random_state=None, tol=0.0001, verbose=0)

In [101]:
label = pd.DataFrame(km.labels_)
label.columns = ['Label']

In [102]:
df3= df2.join(label)

In [103]:
df3.head()

Neighborhood 1st most common venue  \
0                    Regent Park / Harbourfront           Coffee Shop   
1  Queen's Park / Ontario Provincial Government           Coffee Shop   
2                      Garden District, Ryerson           Coffee Shop   
3                                St. James Town             Gastropub   
4                                   The Beaches                 Beach   

  2nd most common venue 3rd most common venue  frequency of first venue   \
0                  Park        Breakfast Spot                          6   
1      Sushi Restaurant    Italian Restaurant                          7   
2               Theater                  Café                          3   
3                  Café        Farmers Market                          3   
4        Breakfast Spot             BBQ Joint                          4   

   frequesncy of 2nd venue  frequency of third venue  Label  
0                        3                         2      2  
1                        3                         2      2  
2                        2                         2      0  
3                        3                         2      0  
4                        3                         2      3

### we have clustered all the neighborhood now lets visualize all cluster on maps

In [106]:
tor =toronto.merge(df3, left_on = 'Neighborhood', right_on = 'Neighborhood', how = 'inner')

In [141]:
tor.head()

Postal code           Borough                                  Neighborhood  \
0         M5A  Downtown Toronto                    Regent Park / Harbourfront   
1         M7A  Downtown Toronto  Queen's Park / Ontario Provincial Government   
2         M5B  Downtown Toronto                      Garden District, Ryerson   
3         M5C  Downtown Toronto                                St. James Town   
4         M4E      East Toronto                                   The Beaches   

    Latitude  Longitude 1st most common venue 2nd most common venue  \
0  43.654260 -79.360636           Coffee Shop                  Park   
1  43.662301 -79.389494           Coffee Shop      Sushi Restaurant   
2  43.657162 -79.378937           Coffee Shop               Theater   
3  43.651494 -79.375418             Gastropub                  Café   
4  43.676357 -79.293031                 Beach        Breakfast Spot   

  3rd most common venue  frequency of first venue   frequesncy of 2nd venue  \
0        Breakfast Spot                          6                        3   
1    Italian Restaurant                          7                        3   
2                  Café                          3                        2   
3        Farmers Market                          3                        3   
4             BBQ Joint                          4                        3   

   frequency of third venue  Label  
0                         2      2  
1                         2      2  
2                         2      0  
3                         2      0  
4                         2      3

In [123]:
from matplotlib import cm
import numpy as np

In [139]:
mymap = folium.Map(location = [tor_lat,tor_long], zoom_start = 10)

clas = tor.Label.unique().shape[0]
color = ['red', 'blue', 'green', 'purple', 'orange']

In [143]:
for col, class_ in zip(list(color),list(tor.Label.unique())):
    tor1 = tor[tor['Label']==class_]
    for lat, long, label in zip(tor1['Latitude'],tor1['Longitude'],tor1['Neighborhood']):
    # add markers 
        labe = folium.Popup(str(label)+" Class-"+(str(class_)), parse_html=True)
        folium.CircleMarker(
            [lat,long],
            radius = 5,
            popup = labe,
            color = col,
            fill=True,
            fill_color=col,
            fill_opacity=0.7).add_to(mymap)
mymap